ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestconnectivity.py

```diff
@@ -39,44 +39,66 @@ def test_definition(self) -> dict | None:
             LOGGER.warning("Variable '%s' is missing from the structured_config. %s is skipped.", str(e), self.__class__.__name__)
             return None
 
-        required_vars = ["type", "ip_address", "peer", "peer_interface"]
+        required_vars = ["name", "peer", "peer_interface", "type", "shutdown"]
 
-        for ethernet_interface in ethernet_interfaces:
+        for idx, ethernet_interface in enumerate(ethernet_interfaces, start=1):
             try:
                 for var in required_vars:
                     get(ethernet_interface, var, required=True)
+            except AristaAvdMissingVariableError as e:
+                LOGGER.warning("Ethernet interface entry #%d from the 'ethernet_interfaces' data model is missing the variable '%s'.", idx, str(e))
+                continue
 
-                peer_ethernet_interfaces = get(self.hostvars, f"{ethernet_interface['peer']}.ethernet_interfaces", required=True)
-                peer_interface = get_item(
-                    peer_ethernet_interfaces,
-                    "name",
-                    ethernet_interface["peer_interface"],
-                    required=True,
-                    var_name=f"name: {ethernet_interface['peer_interface']}",
-                )
-                peer_interface_ip = get(peer_interface, "ip_address", required=True)
-
-                if ethernet_interface["type"] == "routed":
-                    src_ip = str(ip_interface(ethernet_interface["ip_address"]).ip)
-                    dst_ip = str(ip_interface(peer_interface_ip).ip)
-                    custom_field = (
-                        f"Source: {self.device_name}_{ethernet_interface['name']} - Destination:"
-                        f" {ethernet_interface['peer']}_{ethernet_interface['peer_interface']}"
+            if ethernet_interface["type"] == "routed":
+                try:
```

Comment:
continue immediately if not routed
```suggestion
            if ethernet_interface["type"] != "routed":
                continue
                
            try:
```

Reply:
Fixed in 3cb11f738c915058ef7f4d239cdf0da3f1b78f19